helpers/deployer/main.go

```diff
@@ -0,0 +1,145 @@
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	gotest "testing"
+	"time"
+
+	"github.com/mitchellh/go-testing-interface"
+
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/deployer/gcp"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/deployer/steps"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/deployer/stages"
+)
+
+type Cfg struct {
+	tfvarsFile    string
+	stepsFile     string
+	resetStep     string
+	quiet         bool
+	help          bool
+	listSteps     bool
+	disablePrompt bool
+	validate      bool
+	destroy       bool
+}
+
+func validatorApis() []string {
+	return []string{
+		"securitycenter.googleapis.com",
+		"accesscontextmanager.googleapis.com",
+	}
+}
+
+func parseFlags() Cfg {
+	var c Cfg
+
+	flag.StringVar(&c.tfvarsFile, "tfvars_file", "", "Full path to the Terraform .tfvars `file` with the configuration to be used.")
+	flag.StringVar(&c.stepsFile, "steps_file", ".steps.json", "Path to the steps `file` to be used to save progress.")
+	flag.StringVar(&c.resetStep, "reset_step", "", "Name of a `step` to be reset.")
+	flag.BoolVar(&c.quiet, "quiet", false, "If true, additional output is suppressed.")
+	flag.BoolVar(&c.help, "help", false, "Prints this help text and exits.")
+	flag.BoolVar(&c.listSteps, "list_steps", false, "List the existing steps.")
+	flag.BoolVar(&c.disablePrompt, "disable_prompt", false, "Disable interactive prompt.")
+	flag.BoolVar(&c.validate, "validate", false, "Validate tfvars file inputs.")
+	flag.BoolVar(&c.destroy, "destroy", false, "Destroy the deployment.")
+
+	flag.Parse()
+	return c
+}
+
+func main() {
+
+	cfg := parseFlags()
+	if cfg.help {
+		fmt.Println("Deploys the Terraform Example Foundation")
+		flag.PrintDefaults()
+		return
+	}
+
+	// load tfvars
+	globalTfvars, err := stages.ReadGlobalTfvars(cfg.tfvarsFile)
+	if err != nil {
+		fmt.Printf("# Failed to read GlobalTfvars file. Error: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	// validate Directories
+	err = stages.ValidateDirectories(globalTfvars)
+	if err != nil {
+		fmt.Printf("# Failed validating directories. Error: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	// init infra
+	gotest.Init()
+	t := &testing.RuntimeT{}
+
+	// only enable serivices if they are not already enabled
+	if globalTfvars.HasValidatorProj() {
+		var apis []string
+		gcpConf := gcp.NewGCP()
+		for _, a := range validatorApis() {
+			if !gcpConf.ApiIsEnabled(t, *globalTfvars.ValidatorProjectId, a) {
+				apis = append(apis, a)
+			}
+		}
+		if len(apis) > 0 {
+			fmt.Printf("# Enabling APIs: %s in validator project '%s'\n", strings.Join(apis, ", "), *globalTfvars.ValidatorProjectId)
+			gcpConf.EnableApis(t, *globalTfvars.ValidatorProjectId, apis)
+			fmt.Println("# waiting for API propagation")
+			for i := 0; i < 20; i++ {
+				time.Sleep(10 * time.Second)
+				fmt.Println("# waiting for API propagation")
+			}
```

Comment:
the ideia was to give feedback to the user that the helper is still running and is not frozen

```bash
$ ./deployer -tfvars_file="./global.tfvars" -validate
# Enabling APIs: securitycenter.googleapis.com, accesscontextmanager.googleapis.com in validator project 'redacted'
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
# waiting for API propagation
``` 

otherwise we would have only 

```bash
$ ./deployer -tfvars_file="./global.tfvars" -validate
# Enabling APIs: securitycenter.googleapis.com, accesscontextmanager.googleapis.com in validator project 'redacted'
# waiting for API propagation
< 200 seconds of silence >
```

it is the same style of feedback given by Terraform